cpp/src/parquet/arrow/reader.h

```diff
@@ -51,6 +52,43 @@ class ColumnChunkReader;
 class ColumnReader;
 class RowGroupReader;
 
+static constexpr bool DEFAULT_USE_THREADS = false;
+static constexpr bool DEFAULT_READ_DICTIONARY = false;
+
+class PARQUET_EXPORT ArrowReaderProperties {
+ public:
+  explicit ArrowReaderProperties(const bool use_threads = DEFAULT_USE_THREADS,
+                                 bool read_dict = DEFAULT_READ_DICTIONARY)
+      : use_threads_(use_threads), default_read_dict_(read_dict), read_dict_indices_() {}
```

Comment:
If default_read_dict_ is globally configurable (does it need to be?) then add a setter for it

Reply:
I think it didn't make a lot of sense to default read all columns as a dictionary array so I've removed the `default_read_dict_` logic entirely.  I want to flesh this out some more as part of native round-trip for `DictionaryArray` columns.